bioblend/_tests/TestGalaxyJobs.py

```diff
@@ -33,6 +44,57 @@ def test_run_and_rerun_random_lines(self):
         self.assertEqual(build_for_rerun['state_inputs']['seed_source']['seed'], 'asdf')
 
         rerun_output = self.gi.jobs.rerun_job(original_job_id)
-        rerun_output_content = self.gi.datasets.download_dataset(rerun_output['outputs'][0]['id'])
+        rerun_job_id = rerun_output['jobs'][0]['id']
+        self._wait_job(original_job_id)
         original_output_content = self.gi.datasets.download_dataset(original_output['outputs'][0]['id'])
+        self._wait_job(rerun_job_id)
+        rerun_output_content = self.gi.datasets.download_dataset(rerun_output['outputs'][0]['id'])
         self.assertEqual(rerun_output_content, original_output_content)
+
+    @test_util.skip_unless_galaxy('release_21.01')
+    @test_util.skip_unless_tool("Show beginning1")
+    def test_rerun_and_remap(self):
+        path = test_util.get_abspath(os.path.join('data', 'select_first.ga'))
+        wf = self.gi.workflows.import_workflow_from_local_path(path)
+        wf_inputs = {
+            "0": {'src': 'hda', 'id': self.dataset_id},
+            "1": "-1",
+        }
+        inv = self.gi.workflows.invoke_workflow(wf['id'], wf_inputs)
+        for _ in range(120):
+            inv = self.gi.invocations.show_invocation(inv['id'])
+            if inv['steps']:
+                for step in inv['steps']:
+                    if step['job_id']:
+                        self._wait_job(step['job_id'])
+                break
+            time.sleep(.5)
```

Comment:
Probably it's more efficient to wait for the invocation to be scheduled and then wait for the jobs, so maybe wait for the merge of #369 .